browser/ui/views/brave_shields/cookie_list_opt_in_browsertest.cc

```diff
@@ -0,0 +1,188 @@
+  }
+
+  ~CookieListOptInBrowserTest() override = default;
+
+ protected:
+  virtual void SetUpLocalState() { InitializeFilterLists(); }
+
+  void PreRunTestOnMainThread() override {
+    CookieListOptInBubbleHost::AllowBubbleInBackgroundForTesting();
+    SetUpLocalState();
+    InProcessBrowserTest::PreRunTestOnMainThread();
+  }
+
+  auto* regional_service_manager() {
+    return g_brave_browser_process->ad_block_service()
+        ->regional_service_manager();
+  }
+
+  content::WebContents* GetBubbleWebContents() {
+    auto* host = CookieListOptInBubbleHost::FromBrowser(browser());
+    return host ? host->GetBubbleWebContentsForTesting() : nullptr;
+  }
+
+ private:
+  void InitializeFilterLists() {
+    std::vector regional_catalog = {adblock::FilterList(
+        kCookieListUuid,
+        "https://secure.fanboy.co.nz/fanboy-cookiemonster_ubo.txt",
+        "Easylist-Cookie List - Filter Obtrusive Cookie Notices", {},
+        "https://forums.lanik.us/", kRegionalAdBlockComponentTestId,
+        kRegionalAdBlockComponentTest64PublicKey,
+        "Removes obtrusive cookie law notices")};
+    regional_service_manager()->SetRegionalCatalog(regional_catalog);
+  }
+
+  base::test::ScopedFeatureList scoped_feature_list_;
+};
+
+IN_PROC_BROWSER_TEST_F(CookieListOptInBrowserTest, EnableFromBubble) {
+  auto* web_contents = GetBubbleWebContents();
+  ASSERT_TRUE(web_contents);
+
+  base::RunLoop run_loop;
+  PrefChangeRegistrar pref_observer;
+  pref_observer.Init(g_browser_process->local_state());
+  pref_observer.Add(
+      prefs::kAdBlockRegionalFilters,
+      base::BindLambdaForTesting([this, &run_loop]() {
+        if (regional_service_manager()->IsFilterListEnabled(kCookieListUuid)) {
+          run_loop.Quit();
+        }
+      }));
+
+  ASSERT_EQ(true, content::EvalJs(web_contents, R"js(
+    new Promise((resolve) => {
+      setInterval(() => {
+        const elem = document.querySelector('.opt-in-action button')
+        if (elem) {
+          resolve(elem)
+        }
+      }, 30)
+    }).then((elem) => {
+      elem.click()
+      return true
+    })
+  )js"));
+
+  run_loop.Run();
+
+  EXPECT_TRUE(regional_service_manager()->IsFilterListEnabled(kCookieListUuid));
+
+  ASSERT_TRUE(
+      AddTabAtIndex(1, GURL(url::kAboutBlankURL), ui::PAGE_TRANSITION_TYPED));
+
+  WaitForLoadStop(browser()->tab_strip_model()->GetActiveWebContents());
+
+  EXPECT_FALSE(GetBubbleWebContents());
+}
+
+class CookieListOptInPreEnabledBrowserTest : public CookieListOptInBrowserTest {
+ public:
+  ~CookieListOptInPreEnabledBrowserTest() override = default;
+
+ protected:
+  void SetUpLocalState() override {
+    CookieListOptInBrowserTest::SetUpLocalState();
+    regional_service_manager()->EnableFilterList(kCookieListUuid, true);
+  }
+};
+
+IN_PROC_BROWSER_TEST_F(CookieListOptInPreEnabledBrowserTest, AlreadyEnabled) {
+  EXPECT_FALSE(GetBubbleWebContents());
+}
+
+IN_PROC_BROWSER_TEST_F(CookieListOptInBrowserTest, PRE_MultipleWindowRestore) {
```

Comment:
can we move these above `CookieListOptInPreEnabledBrowserTest`? At first glance I thought these were all using the pre-enabled list setting

Reply:
Done.